src/sql/workbench/contrib/notebook/browser/notebookLinkHandler.ts

```diff
@@ -173,9 +174,17 @@ export class NotebookLinkHandler {
  */
 export function findPathRelativeToContent(notebookFolder: string, contentPath: URI | undefined, isMarkdown?: boolean, isEncoded?: boolean): string {
 	if (contentPath?.scheme === 'file') {
-		let relativePath = contentPath.fragment ? path.relative(notebookFolder, contentPath.fsPath).concat('#', contentPath.fragment) : path.relative(notebookFolder, contentPath.fsPath);
+		let decodePath: boolean = contentPath.fsPath !== decodeURI(contentPath.fsPath);
+		let decodedContentPath = contentPath.fsPath;
+		if (decodePath) {
+			decodedContentPath = decodeURI(contentPath.fsPath);
+		}
+		// use the decodedContentPath for calculating the reative path since the notebookFolder is not encoded to get the relative path correctly.
+		let relativePath = contentPath.fragment ? path.relative(notebookFolder, decodedContentPath).concat('#', contentPath.fragment) : path.relative(notebookFolder, decodedContentPath);
 		// if relativePath contains improper directory format due to marked js parsing returning an invalid path (ex. ....\) then we need to replace it to ensure the directories are formatted properly (ex. ..\..\)
 		relativePath = isMarkdown || isEncoded ? replaceInvalidLinkPath(relativePath) : encodeURI(replaceInvalidLinkPath(relativePath)).replace(/%5C/g, '\\');
+		// encode the path if we used decoded contentPath for calculating the relativePath.
+		relativePath = decodePath ? encodeURI(relativePath) : relativePath;
```

Comment:
If this is markdown or isEncoded is true then it's going to double-encode the value...is that what we want?

Reply:
you mean when it's not isMarkdown and not isEncoded ? 
Since I calculated the relativePath from decoded contentPath, I need to encode it before returning. I've had tests failing without this step.

Reply:
Yeah, sorry got that mixed up. 

Are there tests covering the !isMarkdown and !isEncoded paths? Because yes, it makes sense to encode it if you decoded it before but double encoding it definitely doesn't seem like it should result in the correct behavior.

Reply:
Actually all the tests are for !isEncoded and !isMarkdown since we're not setting those attributes. I added a test for covering when those are set as well, thanks!